src/fireedge/src/modules/components/Forms/Vm/AttachPciForm/schema.js

```diff
@@ -0,0 +1,276 @@
+ *
+ * @param {ObjectSchema} pciDevice - PCI device information
+ * @returns {string} - DEVICE, VENDOR and CLASS separated by semicolon
+ */
+export const transformPciToString = (pciDevice = {}) => {
+  const { DEVICE = '', VENDOR = '', CLASS = '' } = pciDevice
+
+  return [DEVICE, VENDOR, CLASS].join(';')
+}
+
+/**
+ * Obtain values from a PCI device String.
+ *
+ * @param {string} pciDevice - DEVICE, VENDOR and CLASS separated by semicolon
+ * @returns {ObjectSchema} - PCI device information
+ */
+export const getPciAttributes = (pciDevice = '') => {
+  const [DEVICE, VENDOR, CLASS] = pciDevice.split(';')
+
+  return { DEVICE, VENDOR, CLASS }
+}
+
+/** @type {Field} Specific device PCI device field */
+const SPECIFIC_DEVICE = {
+  name: 'SPECIFIC_DEVICE',
+  label: T.PCISpecificDevice,
+  tooltip: T.PCISpecificDeviceHelp,
+  type: INPUT_TYPES.SWITCH,
+  validation: boolean().notRequired(),
+  grid: { sm: 12, md: 12 },
+}
+
+/** @type {Field} Name PCI device field */
+const NAME_FIELD = {
+  name: 'PCI_DEVICE_NAME',
+  label: T.DeviceName,
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  optionsOnly: true,
+  dependOf: SPECIFIC_DEVICE.name,
+  htmlType: (specificDevice) => specificDevice && INPUT_TYPES.HIDDEN,
+  values: () => {
+    const { data: hosts = [] } = HostAPI.useGetHostsAdminQuery()
+    const pciDevices = hosts.map(getPciDevices).flat()
+
+    return arrayToOptions(uniqWith(pciDevices, samePciDevice), {
+      getText: ({ DEVICE_NAME } = {}) => DEVICE_NAME,
+      getValue: transformPciToString,
+    })
+  },
+  validation: string()
+    .trim()
+    .notRequired()
+    .afterSubmit(() => undefined),
+  grid: { sm: 12, md: 4 },
+}
+
+/** @type {Field} PCI device field */
+const DEVICE_FIELD = {
+  name: 'DEVICE',
+  label: T.Device,
+  type: INPUT_TYPES.TEXT,
+  dependOf: [NAME_FIELD.name, SPECIFIC_DEVICE.name],
+  htmlType: ([_, specificDevice] = []) => specificDevice && INPUT_TYPES.HIDDEN,
+  watcher: ([pciDevice] = []) => {
+    if (pciDevice) {
+      const { DEVICE: attribute } = getPciAttributes(pciDevice)
+
+      return attribute
+    }
+  },
+  fieldProps: { disabled: true },
+  validation: string()
+    .trim()
+    .when(SPECIFIC_DEVICE.name, (type, schema) => {
+      schema.afterSubmit((value, { context }) =>
+        context?.SPECIFIC_DEVICE ? undefined : value
+      )
+
+      return type ? schema.notRequired() : schema.required()
+    }),
+  grid: { xs: 12, md: 2 },
+}
+
+/** @type {Field} PCI device field */
+const VENDOR_FIELD = {
+  name: 'VENDOR',
+  label: T.Vendor,
+  type: INPUT_TYPES.TEXT,
+  dependOf: [NAME_FIELD.name, SPECIFIC_DEVICE.name],
+  htmlType: ([_, specificDevice] = []) => specificDevice && INPUT_TYPES.HIDDEN,
+  watcher: ([pciDevice] = []) => {
+    if (pciDevice) {
+      const { VENDOR: attribute } = getPciAttributes(pciDevice)
+
+      return attribute
+    }
+  },
+  fieldProps: { disabled: true },
+  validation: string()
+    .trim()
+    .when(SPECIFIC_DEVICE.name, (type, schema) => {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).